signals.go

```diff
@@ -17,6 +18,10 @@ const signalBufferSize = 2048
 // newSignalHandler returns a signal handler for processing SIGCHLD and SIGWINCH signals
 // while still forwarding all other signals to the process.
 func newSignalHandler(tty *tty) *signalHandler {
+	// set us as the subpreaper before registering the signal handler for the container
+	if err := system.SetSubreaper(1); err != nil {
```

Comment:
It is needed to do a wait if you are running in PID host for grandchild processes of the container.  I think it is a safe thing to set but we could have a flag or something else.